modules/openy_features/openy_prgf/modules/openy_prgf_class_sessions/openy_prgf_class_sessions.module

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * @file OpenY Paragraph Class Sessions module file.
+ */
+
+use Drupal\openy_session_instance\Entity\SessionInstance;
+use Drupal\openy_session_instance\SessionInstanceManager;
+use Drupal\node\Entity\Node;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_theme().
+ */
+function openy_prgf_class_theme() {
+  $items = [
+    'class_sessions' => [
+      'variables' => [
+        'session_instances' => [],
+      ],
+      'template' => 'openy-class-sessions',
+    ],
+  ];
+  return $items;
+}
+
+/**
+ * Implements hook_preprocess_paragraph__bundle().
+ */
+function openy_prgf_class_sessions_preprocess_paragraph__class_sessions(&$variables) {
```

Comment:
1. We'll use OOP and know where to find code for block itself. Along with that we'll have less custom code.
2. With block we provide support not only for paragraphs, because anytime this block may be placed without paragraph.
3. If later there will be added Panelized or Panels - with block, we'll be able to placed block instead of using paragraphs.
4. Using preprocess is old school way. We should avoid this in Drupal 8.